action/action.go

```diff
@@ -18,7 +18,7 @@ import (
 type (
 	// Action is the action can be Executed in protocols. The method is added to avoid mistakenly used empty interface as action.
 	Action interface {
-		SetEnvelopeContext(SealedEnvelope)
+		SetAbstractAction(AbstractAction)
```

Comment:
Why is `SetAbstractAction` needed in `Action` interface?

Reply:
good catch, it can be removed

Reply:
yes, removed

Reply:
remove